src/webapi/element.rs

```diff
@@ -163,8 +163,46 @@ pub trait IElement: INode + IParentNode + IChildNode {
     fn has_pointer_capture( &self, pointer_id: i32 ) -> bool {
         js!( return @{self.as_ref()}.hasPointerCapture( @{pointer_id} ); ).try_into().unwrap()
     }
+
+    /// Insert nodes from HTML fragment into specified position.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML)
+    fn insert_adjacent_html( &self, position: &str, html: &str ) -> Result<(), SyntaxError> {
+        js_try!( @(no_return)
+            @{self.as_ref()}.insertAdjacentHTML( @{position}, @{html} );
+        ).unwrap()
+    }
+
+    /// Insert nodes from HTML fragment before element.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML)
+    fn insert_html_before_begin( &self, html: &str ) -> Result<(), SyntaxError> {
+        self.insert_adjacent_html("beforebegin", html)
+    }
+
+    /// Insert nodes from HTML fragment as the first children of the element.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML)
+    fn insert_html_after_begin( &self, html: &str ) -> Result<(), SyntaxError> {
+        self.insert_adjacent_html("afterbegin", html)
+    }
+
+    /// Insert nodes from HTML fragment as the last children of the element.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML)
+    fn insert_html_before_end( &self, html: &str ) -> Result<(), SyntaxError> {
+        self.insert_adjacent_html("beforeend", html)
+    }
+
+    /// Insert nodes from HTML fragment after element.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML)
+    fn insert_html_after_end( &self, html: &str ) -> Result<(), SyntaxError> {
```

Comment:
Hmmm.. are those necessary?

I guess I'm not entirely against having them for conveniance, however if we're going to invent new methods I'd rather have them named better than the original JS one. (e.g. maybe `append_html`, `prepend_html`, `append_html_outside`, `prepend_html_outside`, or something like that?)

Reply:
They aren't very necessary but they save a lot of typing and show up in autocomplete lists which I find very useful. I would prefer to have them.

As for the names I'm not the biggest fan either. I like `prepend_html` and `append_html`. However I don't really like `append_html_outside` and `append_html_outside`. The words append/prepend don't seem to fit when the content is likely going somewhere in the middle of the parent element.

Reply:
> However I don't really like `append_html_outside` and `prepend_html_outside`. The words append/prepend don't seem to fit when the content is likely going somewhere in the middle of the parent element.

Hmmm... then how about `insert_html_before` and `insert_html_after`?

Reply:
I like those. I will update to `prepend_html`, `append_html`, `insert_html_before` and `insert_html_after`.

Reply:
Sounds good to me!

Reply:
Ping? (:

Reply:
Done.